src/prop/bvminisat/core/Solver.cc

```diff
@@ -1482,11 +1483,11 @@ void ClauseAllocator::reloc(CRef& cr,
 }
 
 void Solver::setNotify(Notify* toNotify) { d_notify = toNotify; }
-bool Solver::withinBudget(uint64_t amount) const
+bool Solver::withinBudget(ResourceManager::Resource r) const
 {
   AlwaysAssert(d_notify);
-  d_notify->spendResource(amount);
-  d_notify->safePoint(0);
+  d_notify->spendResource(r);
+  d_notify->safePoint(r);
```

Comment:
Shouldn't the two lines above be just `d_notify->safePoint(r);` since `safePoint()` calls `spendResource` and the original code did not spend the `amount` twice.

Reply:
Yes you are right. `safePoint()` will call `spendResource` in https://github.com/CVC4/CVC4/blob/master/src/theory/theory_engine.h#L284 . I wonder why it was not `d_notify->safePoint(amount);` from the beginning.